android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -649,12 +650,28 @@ constructor(
     questionnaire: Questionnaire,
     subjectId: String,
     subjectType: ResourceType,
+    questionnaireConfig: QuestionnaireConfig,
   ): QuestionnaireResponse {
-    var questionnaireResponse =
-      loadQuestionnaireResponse(subjectId, subjectType, questionnaire.logicalId)
-
+    var questionnaireResponse: QuestionnaireResponse? = null
+    // if questionnaireType is Default that means we don't have to load response from DB
+    if (!questionnaireConfig.type.isDefault()) {
+      questionnaireResponse =
+        loadQuestionnaireResponse(subjectId, subjectType, questionnaire.logicalId)
+    }
+    var populationResources = ArrayList<Resource>()
     if (questionnaireResponse == null) {
-      val populationResources = loadPopulationResources(subjectId, subjectType)
+      if (subjectType == ResourceType.Task) {
+        if (questionnaireConfig.resourceIdentifier != null &&
+            questionnaireConfig.resourceType != null
+        ) {
+          populationResources =
+            loadPopulationResources(
+              questionnaireConfig.resourceIdentifier!!,
+              questionnaireConfig.resourceType!!
+            )
+        }
+      }
```

Comment:
@aurangzaibumer looks like the previous implementation specifically tested against Tasks as a base resource being supported with the test case `testQRFromPopulationWithTaskBaseResourceTypeIsNotSupported`, but this PR implements it. 

We need need to refactor this to either have a unit test that support QR from population for Tasks or revert the changes specific to this update in order to fix the build.

Reply:
Do we need that check on Line 663?

Reply:
if subjectType is Patient, we won't be repopulating it twice or if the subjectType is Group, we don't have to populate it for that too. That is why I did add this check.

Reply:
I think we can apply negation instead, then update `loadPopulationResources` to work with other resource types (not just the `Task` resource). My assumption: the function should be used to populate any resource type.

Reply:
Suggested refactor for `loadPopulationResources`

```kotlin

 /**
   * Loads resources to be populated into a Questionnaire Response.
   *
   * @param subjectId resource uuid
   * @param subjectType resource type
   */
  private suspend fun loadPopulationResources(
    subjectId: String,
    subjectType: ResourceType
  ): ArrayList<Resource> {
    val populationResources = arrayListOf<Resource>()
    populationResources.add(defaultRepository.loadResource(subjectId, subjectType))
    if (subjectType == ResourceType.Patient) {
        loadRelatedPerson(subjectId)?.run { populationResources.add(this) }
    }
    return populationResources
  }
````


`Patient` resource is a special case here that requires loading of 2 resource types (the `Patient` and `RelatedPerson`) resources.  

Then delete functions `loadTask` and `loadGroup` as they are no longer needed.